scripts/distributeVersion/distributeVersion.ts

```diff
@@ -0,0 +1,174 @@
+// Update TypeScript Version
+
+/// <reference path="..\typings\node\node.d.ts" />
+
+import * as fs from "fs";
+import * as path from "path";
+
+let versionFilePath: string;
+
+interface FileInformation {
+    filePath: string;
+    ext: string;
+    content: string;
+}
+
+interface FormatInformation {
+    pattern: RegExp;
+    serialize: (version: string) => string;
+}
+
+/**
+ * A minimal description for a parsed package.json object.
+ */
+interface VersionJson {
+    version: string[];
+}
+
+/**
+ * A map between filePath and FormatInformation.
+ * A new file that need to be serialize and update version should be added in this map.
+ */
+const formats: {[idx: string]: FormatInformation} = {
+    "src/compiler/program.ts": {
+        pattern: /const\s*version\s*=\s*"[0-9]+\.[0-9]+\.[0-9]"/g,
+        serialize: (version: string) => {
+            return `const version = "${version}"`;
+        }
+    },
+    "package.json": {
+        pattern: /"version"\s*:\s*"[0-9]+\.[0-9]+\.[0-9]"/g,
+        serialize: (version: string) => {
+            return `"version": "${version}"`;
+        }
+    }
+};
+
+function getFileInformationSync(filePath: string): FileInformation {
+    try {
+        const data = fs.readFileSync(filePath, "utf8");
+        return {
+            filePath,
+            content: data,
+            ext: path.extname(filePath)
+        };
+    }
+    catch (error) {
+        throw new Error("Error in getFileInformationSync: " + error);
+    }
+
+}
+
+function getVersionSync(): string {
+    try {
+        const packageJson: VersionJson = JSON.parse(fs.readFileSync(versionFilePath, "utf8"));
+        // The version has 4 digits, we only need 3
+        const version = packageJson.version;
+        return version.slice(0, 3).join(".");

```

Comment:
Move these out of your `try` since they shouldn't fail.

Then throw an error if the version number doesn't match the expected format.